<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ol.lst-kix_90oyjc4qrwtp-2.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-2 0}.lst-kix_51wdrdk2fq9o-2>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-2}.lst-kix_qwuufa9dxghx-5>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-5}.lst-kix_pqruqr21jqkv-3>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-3}ol.lst-kix_epldur8jqq61-0.start{counter-reset:lst-ctn-kix_epldur8jqq61-0 0}ol.lst-kix_51wdrdk2fq9o-2.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-2 0}ol.lst-kix_3qvy38k9i9l4-7.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-7 0}.lst-kix_xf4ne1qm8zsq-4>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-4}.lst-kix_2d4m39kn8zc9-7>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-7,lower-latin) ". "}ol.lst-kix_li21gc20orp9-3.start{counter-reset:lst-ctn-kix_li21gc20orp9-3 0}ol.lst-kix_2d4m39kn8zc9-6.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-6 0}.lst-kix_g65pgrp6l07-5>li{counter-increment:lst-ctn-kix_g65pgrp6l07-5}.lst-kix_2d4m39kn8zc9-8>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-8,lower-roman) ". "}ol.lst-kix_pqruqr21jqkv-5.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-5 0}ol.lst-kix_xf4ne1qm8zsq-5.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-5 0}ol.lst-kix_3qvy38k9i9l4-2.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-2 0}.lst-kix_bc3xa550q60r-2>li{counter-increment:lst-ctn-kix_bc3xa550q60r-2}ol.lst-kix_qwuufa9dxghx-0.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-0 0}ol.lst-kix_bc3xa550q60r-0{list-style-type:none}ol.lst-kix_bc3xa550q60r-2{list-style-type:none}ol.lst-kix_bc3xa550q60r-1{list-style-type:none}ol.lst-kix_bc3xa550q60r-8{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-0.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-0 0}ol.lst-kix_bc3xa550q60r-7{list-style-type:none}ol.lst-kix_bc3xa550q60r-4{list-style-type:none}ol.lst-kix_bc3xa550q60r-3{list-style-type:none}ol.lst-kix_bc3xa550q60r-6{list-style-type:none}ol.lst-kix_g65pgrp6l07-4.start{counter-reset:lst-ctn-kix_g65pgrp6l07-4 0}ol.lst-kix_bc3xa550q60r-5{list-style-type:none}ol.lst-kix_epldur8jqq61-5.start{counter-reset:lst-ctn-kix_epldur8jqq61-5 0}.lst-kix_2d4m39kn8zc9-7>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-7}.lst-kix_hedhuo5tdgo-4>li:before{content:"-  "}.lst-kix_hedhuo5tdgo-5>li:before{content:"-  "}.lst-kix_hedhuo5tdgo-6>li:before{content:"-  "}.lst-kix_2d4m39kn8zc9-4>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-4,lower-latin) ". "}.lst-kix_hedhuo5tdgo-7>li:before{content:"-  "}.lst-kix_hedhuo5tdgo-8>li:before{content:"-  "}.lst-kix_2d4m39kn8zc9-5>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-5,lower-roman) ". "}.lst-kix_2d4m39kn8zc9-6>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-6,decimal) ". "}ol.lst-kix_52z049lpujuv-2.start{counter-reset:lst-ctn-kix_52z049lpujuv-2 0}ol.lst-kix_li21gc20orp9-8.start{counter-reset:lst-ctn-kix_li21gc20orp9-8 0}ol.lst-kix_pqruqr21jqkv-3{list-style-type:none}ol.lst-kix_pqruqr21jqkv-4{list-style-type:none}ol.lst-kix_pqruqr21jqkv-1{list-style-type:none}ol.lst-kix_pqruqr21jqkv-2{list-style-type:none}.lst-kix_2d4m39kn8zc9-3>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-3,decimal) ". "}ol.lst-kix_pqruqr21jqkv-7{list-style-type:none}ol.lst-kix_pqruqr21jqkv-8{list-style-type:none}ol.lst-kix_pqruqr21jqkv-5{list-style-type:none}ol.lst-kix_pqruqr21jqkv-6{list-style-type:none}.lst-kix_2d4m39kn8zc9-2>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-2,lower-roman) ". "}.lst-kix_52z049lpujuv-3>li{counter-increment:lst-ctn-kix_52z049lpujuv-3}.lst-kix_rcus5r4ke429-0>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-0,decimal) ". "}.lst-kix_2d4m39kn8zc9-1>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-1,lower-latin) ". "}.lst-kix_rcus5r4ke429-1>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-1,lower-latin) ". "}ol.lst-kix_pqruqr21jqkv-0{list-style-type:none}.lst-kix_g65pgrp6l07-1>li{counter-increment:lst-ctn-kix_g65pgrp6l07-1}.lst-kix_2d4m39kn8zc9-0>li:before{content:"" counter(lst-ctn-kix_2d4m39kn8zc9-0,decimal) ". "}.lst-kix_epldur8jqq61-4>li{counter-increment:lst-ctn-kix_epldur8jqq61-4}ol.lst-kix_qwuufa9dxghx-5.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-5 0}.lst-kix_90oyjc4qrwtp-5>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-5}ol.lst-kix_bc3xa550q60r-2.start{counter-reset:lst-ctn-kix_bc3xa550q60r-2 0}.lst-kix_rcus5r4ke429-6>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-6,decimal) ". "}.lst-kix_rcus5r4ke429-8>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-8,lower-roman) ". "}ul.lst-kix_642xdqsnfsyh-0{list-style-type:none}ul.lst-kix_642xdqsnfsyh-2{list-style-type:none}ul.lst-kix_642xdqsnfsyh-1{list-style-type:none}.lst-kix_rcus5r4ke429-2>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-2,lower-roman) ". "}.lst-kix_rcus5r4ke429-4>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-4,lower-latin) ". "}ul.lst-kix_642xdqsnfsyh-4{list-style-type:none}ul.lst-kix_642xdqsnfsyh-3{list-style-type:none}ul.lst-kix_642xdqsnfsyh-6{list-style-type:none}ul.lst-kix_642xdqsnfsyh-5{list-style-type:none}ol.lst-kix_bc3xa550q60r-4.start{counter-reset:lst-ctn-kix_bc3xa550q60r-4 0}ul.lst-kix_642xdqsnfsyh-8{list-style-type:none}ul.lst-kix_642xdqsnfsyh-7{list-style-type:none}ol.lst-kix_li21gc20orp9-5.start{counter-reset:lst-ctn-kix_li21gc20orp9-5 0}.lst-kix_epldur8jqq61-8>li{counter-increment:lst-ctn-kix_epldur8jqq61-8}ol.lst-kix_90oyjc4qrwtp-8{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-7{list-style-type:none}ol.lst-kix_rcus5r4ke429-4.start{counter-reset:lst-ctn-kix_rcus5r4ke429-4 0}ol.lst-kix_bc3xa550q60r-7.start{counter-reset:lst-ctn-kix_bc3xa550q60r-7 0}ol.lst-kix_90oyjc4qrwtp-0{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-2{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-1{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-4{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-3{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-6{list-style-type:none}ol.lst-kix_qwuufa9dxghx-7.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-7 0}ol.lst-kix_90oyjc4qrwtp-5{list-style-type:none}ol.lst-kix_52z049lpujuv-4.start{counter-reset:lst-ctn-kix_52z049lpujuv-4 0}.lst-kix_hedhuo5tdgo-3>li:before{content:"-  "}.lst-kix_hedhuo5tdgo-1>li:before{content:"-  "}.lst-kix_3qvy38k9i9l4-3>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-3}ol.lst-kix_epldur8jqq61-2.start{counter-reset:lst-ctn-kix_epldur8jqq61-2 0}.lst-kix_qwuufa9dxghx-2>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-2}.lst-kix_51wdrdk2fq9o-5>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-5}ol.lst-kix_g65pgrp6l07-1.start{counter-reset:lst-ctn-kix_g65pgrp6l07-1 0}.lst-kix_p5y3sb60oxh6-1>li:before{content:"-  "}.lst-kix_bc3xa550q60r-3>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-3,decimal) ". "}.lst-kix_bc3xa550q60r-7>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-7,lower-latin) ". "}ol.lst-kix_90oyjc4qrwtp-7.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-7 0}.lst-kix_p5y3sb60oxh6-3>li:before{content:"-  "}.lst-kix_qwuufa9dxghx-8>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-8}.lst-kix_xf4ne1qm8zsq-8>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-8}.lst-kix_li21gc20orp9-8>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-8,lower-roman) ". "}.lst-kix_bc3xa550q60r-1>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-1,lower-latin) ". "}ol.lst-kix_pqruqr21jqkv-8.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-8 0}.lst-kix_li21gc20orp9-2>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-2,lower-roman) ". "}.lst-kix_li21gc20orp9-6>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-6,decimal) ". "}.lst-kix_qwuufa9dxghx-1>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-1}ol.lst-kix_52z049lpujuv-7.start{counter-reset:lst-ctn-kix_52z049lpujuv-7 0}.lst-kix_51wdrdk2fq9o-6>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-6}ol.lst-kix_51wdrdk2fq9o-0.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-0 0}.lst-kix_li21gc20orp9-5>li{counter-increment:lst-ctn-kix_li21gc20orp9-5}.lst-kix_pqruqr21jqkv-0>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-0,decimal) ". "}.lst-kix_rcus5r4ke429-2>li{counter-increment:lst-ctn-kix_rcus5r4ke429-2}.lst-kix_li21gc20orp9-4>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-4,lower-latin) ". "}.lst-kix_bc3xa550q60r-5>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-5,lower-roman) ". "}.lst-kix_3qvy38k9i9l4-4>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-4}ol.lst-kix_2d4m39kn8zc9-1.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-1 0}.lst-kix_pqruqr21jqkv-6>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-6,decimal) ". "}.lst-kix_2d4m39kn8zc9-6>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-6}ol.lst-kix_3qvy38k9i9l4-4.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-4 0}.lst-kix_pqruqr21jqkv-2>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-2,lower-roman) ". "}.lst-kix_2d4m39kn8zc9-0>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-0}ol.lst-kix_rcus5r4ke429-2.start{counter-reset:lst-ctn-kix_rcus5r4ke429-2 0}.lst-kix_pqruqr21jqkv-4>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-4,lower-latin) ". "}.lst-kix_li21gc20orp9-0>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-0,decimal) ". "}.lst-kix_p5y3sb60oxh6-7>li:before{content:"-  "}.lst-kix_p5y3sb60oxh6-5>li:before{content:"-  "}ol.lst-kix_pqruqr21jqkv-7.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-7 0}.lst-kix_pqruqr21jqkv-8>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-8,lower-roman) ". "}.lst-kix_xf4ne1qm8zsq-1>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-1,lower-latin) ". "}.lst-kix_xf4ne1qm8zsq-2>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-2,lower-roman) ". "}.lst-kix_xf4ne1qm8zsq-5>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-5,lower-roman) ". "}ol.lst-kix_51wdrdk2fq9o-2{list-style-type:none}.lst-kix_52z049lpujuv-2>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-2,lower-roman) ". "}ol.lst-kix_51wdrdk2fq9o-3{list-style-type:none}ol.lst-kix_51wdrdk2fq9o-0{list-style-type:none}.lst-kix_qwuufa9dxghx-4>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-4}ol.lst-kix_51wdrdk2fq9o-1{list-style-type:none}.lst-kix_xf4ne1qm8zsq-6>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-6,decimal) ". "}ol.lst-kix_52z049lpujuv-5.start{counter-reset:lst-ctn-kix_52z049lpujuv-5 0}.lst-kix_3qvy38k9i9l4-8>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-8}.lst-kix_51wdrdk2fq9o-3>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-3}ol.lst-kix_2d4m39kn8zc9-3.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-3 0}.lst-kix_52z049lpujuv-3>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-3,decimal) ". "}ol.lst-kix_g65pgrp6l07-2.start{counter-reset:lst-ctn-kix_g65pgrp6l07-2 0}ol.lst-kix_epldur8jqq61-3.start{counter-reset:lst-ctn-kix_epldur8jqq61-3 0}ol.lst-kix_qwuufa9dxghx-4{list-style-type:none}ol.lst-kix_qwuufa9dxghx-5{list-style-type:none}.lst-kix_li21gc20orp9-8>li{counter-increment:lst-ctn-kix_li21gc20orp9-8}ol.lst-kix_qwuufa9dxghx-6{list-style-type:none}ol.lst-kix_qwuufa9dxghx-7{list-style-type:none}ol.lst-kix_bc3xa550q60r-0.start{counter-reset:lst-ctn-kix_bc3xa550q60r-0 0}ol.lst-kix_qwuufa9dxghx-8{list-style-type:none}ol.lst-kix_g65pgrp6l07-7.start{counter-reset:lst-ctn-kix_g65pgrp6l07-7 0}ol.lst-kix_2d4m39kn8zc9-8.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-8 0}.lst-kix_52z049lpujuv-6>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-6,decimal) ". "}ol.lst-kix_qwuufa9dxghx-0{list-style-type:none}ol.lst-kix_qwuufa9dxghx-1{list-style-type:none}.lst-kix_52z049lpujuv-7>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-7,lower-latin) ". "}ol.lst-kix_qwuufa9dxghx-2{list-style-type:none}ol.lst-kix_qwuufa9dxghx-3{list-style-type:none}.lst-kix_li21gc20orp9-6>li{counter-increment:lst-ctn-kix_li21gc20orp9-6}.lst-kix_3qvy38k9i9l4-6>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-6}ol.lst-kix_g65pgrp6l07-8.start{counter-reset:lst-ctn-kix_g65pgrp6l07-8 0}ol.lst-kix_li21gc20orp9-0.start{counter-reset:lst-ctn-kix_li21gc20orp9-0 0}.lst-kix_2d4m39kn8zc9-8>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-8}ol.lst-kix_rcus5r4ke429-0.start{counter-reset:lst-ctn-kix_rcus5r4ke429-0 0}.lst-kix_xf4ne1qm8zsq-1>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-1}.lst-kix_pqruqr21jqkv-6>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-6}.lst-kix_rcus5r4ke429-3>li{counter-increment:lst-ctn-kix_rcus5r4ke429-3}ol.lst-kix_51wdrdk2fq9o-8{list-style-type:none}.lst-kix_epldur8jqq61-1>li{counter-increment:lst-ctn-kix_epldur8jqq61-1}.lst-kix_qwuufa9dxghx-6>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-6}ol.lst-kix_51wdrdk2fq9o-6{list-style-type:none}ol.lst-kix_51wdrdk2fq9o-7{list-style-type:none}ol.lst-kix_51wdrdk2fq9o-4{list-style-type:none}ol.lst-kix_51wdrdk2fq9o-5{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-4.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-4 0}.lst-kix_g65pgrp6l07-4>li{counter-increment:lst-ctn-kix_g65pgrp6l07-4}.lst-kix_52z049lpujuv-6>li{counter-increment:lst-ctn-kix_52z049lpujuv-6}ol.lst-kix_li21gc20orp9-1.start{counter-reset:lst-ctn-kix_li21gc20orp9-1 0}ol.lst-kix_g65pgrp6l07-3.start{counter-reset:lst-ctn-kix_g65pgrp6l07-3 0}.lst-kix_51wdrdk2fq9o-1>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-1}.lst-kix_90oyjc4qrwtp-2>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-2}ol.lst-kix_3qvy38k9i9l4-0.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-0 0}ul.lst-kix_p5y3sb60oxh6-1{list-style-type:none}ol.lst-kix_g65pgrp6l07-6.start{counter-reset:lst-ctn-kix_g65pgrp6l07-6 0}ul.lst-kix_p5y3sb60oxh6-0{list-style-type:none}.lst-kix_52z049lpujuv-8>li{counter-increment:lst-ctn-kix_52z049lpujuv-8}.lst-kix_rcus5r4ke429-5>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-5,lower-roman) ". "}.lst-kix_epldur8jqq61-6>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-6,decimal) ". "}.lst-kix_li21gc20orp9-1>li{counter-increment:lst-ctn-kix_li21gc20orp9-1}.lst-kix_90oyjc4qrwtp-6>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-6,decimal) ". "}.lst-kix_90oyjc4qrwtp-2>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-2,lower-roman) ". "}.lst-kix_rcus5r4ke429-5>li{counter-increment:lst-ctn-kix_rcus5r4ke429-5}ol.lst-kix_bc3xa550q60r-1.start{counter-reset:lst-ctn-kix_bc3xa550q60r-1 0}.lst-kix_90oyjc4qrwtp-0>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-0}.lst-kix_bc3xa550q60r-7>li{counter-increment:lst-ctn-kix_bc3xa550q60r-7}.lst-kix_g65pgrp6l07-2>li{counter-increment:lst-ctn-kix_g65pgrp6l07-2}.lst-kix_bc3xa550q60r-5>li{counter-increment:lst-ctn-kix_bc3xa550q60r-5}.lst-kix_li21gc20orp9-3>li{counter-increment:lst-ctn-kix_li21gc20orp9-3}.lst-kix_hedhuo5tdgo-2>li:before{content:"-  "}.lst-kix_qwuufa9dxghx-5>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-5,lower-roman) ". "}ol.lst-kix_52z049lpujuv-1.start{counter-reset:lst-ctn-kix_52z049lpujuv-1 0}.lst-kix_51wdrdk2fq9o-8>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-8}ol.lst-kix_epldur8jqq61-7.start{counter-reset:lst-ctn-kix_epldur8jqq61-7 0}.lst-kix_51wdrdk2fq9o-7>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-7,lower-latin) ". "}.lst-kix_bc3xa550q60r-6>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-6,decimal) ". "}.lst-kix_p5y3sb60oxh6-2>li:before{content:"-  "}.lst-kix_642xdqsnfsyh-7>li:before{content:"-  "}.lst-kix_li21gc20orp9-7>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-7,lower-latin) ". "}.lst-kix_bc3xa550q60r-2>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-2,lower-roman) ". "}.lst-kix_epldur8jqq61-6>li{counter-increment:lst-ctn-kix_epldur8jqq61-6}.lst-kix_642xdqsnfsyh-3>li:before{content:"-  "}.lst-kix_li21gc20orp9-3>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-3,decimal) ". "}.lst-kix_51wdrdk2fq9o-3>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-3,decimal) ". "}ol.lst-kix_epldur8jqq61-8.start{counter-reset:lst-ctn-kix_epldur8jqq61-8 0}.lst-kix_3qvy38k9i9l4-4>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-4,lower-latin) ". "}.lst-kix_3qvy38k9i9l4-1>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-1}.lst-kix_2d4m39kn8zc9-3>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-3}.lst-kix_3qvy38k9i9l4-8>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-8,lower-roman) ". "}ol.lst-kix_52z049lpujuv-0.start{counter-reset:lst-ctn-kix_52z049lpujuv-0 0}.lst-kix_pqruqr21jqkv-3>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-3,decimal) ". "}.lst-kix_p5y3sb60oxh6-6>li:before{content:"-  "}.lst-kix_pqruqr21jqkv-8>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-8}ol.lst-kix_2d4m39kn8zc9-1{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-2{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-0{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-5{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-6{list-style-type:none}ul.lst-kix_p5y3sb60oxh6-8{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-3{list-style-type:none}ul.lst-kix_p5y3sb60oxh6-7{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-4{list-style-type:none}ul.lst-kix_p5y3sb60oxh6-6{list-style-type:none}ul.lst-kix_p5y3sb60oxh6-5{list-style-type:none}.lst-kix_pqruqr21jqkv-7>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-7,lower-latin) ". "}ul.lst-kix_p5y3sb60oxh6-4{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-7{list-style-type:none}ul.lst-kix_p5y3sb60oxh6-3{list-style-type:none}ol.lst-kix_2d4m39kn8zc9-7.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-7 0}ol.lst-kix_2d4m39kn8zc9-8{list-style-type:none}ul.lst-kix_p5y3sb60oxh6-2{list-style-type:none}ol.lst-kix_52z049lpujuv-8.start{counter-reset:lst-ctn-kix_52z049lpujuv-8 0}.lst-kix_2d4m39kn8zc9-5>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-5}ol.lst-kix_rcus5r4ke429-8.start{counter-reset:lst-ctn-kix_rcus5r4ke429-8 0}.lst-kix_li21gc20orp9-7>li{counter-increment:lst-ctn-kix_li21gc20orp9-7}.lst-kix_bc3xa550q60r-0>li{counter-increment:lst-ctn-kix_bc3xa550q60r-0}ol.lst-kix_2d4m39kn8zc9-0.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-0 0}ol.lst-kix_bc3xa550q60r-8.start{counter-reset:lst-ctn-kix_bc3xa550q60r-8 0}.lst-kix_g65pgrp6l07-7>li{counter-increment:lst-ctn-kix_g65pgrp6l07-7}ol.lst-kix_qwuufa9dxghx-6.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-6 0}.lst-kix_90oyjc4qrwtp-7>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-7}.lst-kix_epldur8jqq61-2>li{counter-increment:lst-ctn-kix_epldur8jqq61-2}.lst-kix_xf4ne1qm8zsq-6>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-6}.lst-kix_3qvy38k9i9l4-3>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-3,decimal) ". "}.lst-kix_3qvy38k9i9l4-2>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-2,lower-roman) ". "}.lst-kix_3qvy38k9i9l4-1>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-1,lower-latin) ". "}ol.lst-kix_g65pgrp6l07-5.start{counter-reset:lst-ctn-kix_g65pgrp6l07-5 0}ol.lst-kix_51wdrdk2fq9o-8.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-8 0}.lst-kix_3qvy38k9i9l4-0>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-0,decimal) ". "}.lst-kix_52z049lpujuv-1>li{counter-increment:lst-ctn-kix_52z049lpujuv-1}.lst-kix_pqruqr21jqkv-1>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-1}ol.lst-kix_52z049lpujuv-3.start{counter-reset:lst-ctn-kix_52z049lpujuv-3 0}.lst-kix_qwuufa9dxghx-4>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-4,lower-latin) ". "}.lst-kix_qwuufa9dxghx-3>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-3,decimal) ". "}.lst-kix_qwuufa9dxghx-2>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-2,lower-roman) ". "}.lst-kix_g65pgrp6l07-0>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-0,decimal) ". "}ol.lst-kix_2d4m39kn8zc9-5.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-5 0}.lst-kix_qwuufa9dxghx-0>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-0,decimal) ". "}.lst-kix_qwuufa9dxghx-1>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-1,lower-latin) ". "}.lst-kix_g65pgrp6l07-1>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-1,lower-latin) ". "}.lst-kix_g65pgrp6l07-2>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-2,lower-roman) ". "}ol.lst-kix_bc3xa550q60r-3.start{counter-reset:lst-ctn-kix_bc3xa550q60r-3 0}ol.lst-kix_3qvy38k9i9l4-1.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-1 0}.lst-kix_bc3xa550q60r-4>li{counter-increment:lst-ctn-kix_bc3xa550q60r-4}.lst-kix_2d4m39kn8zc9-1>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-1}.lst-kix_3qvy38k9i9l4-5>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-5}ol.lst-kix_epldur8jqq61-6.start{counter-reset:lst-ctn-kix_epldur8jqq61-6 0}.lst-kix_g65pgrp6l07-5>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-5,lower-roman) ". "}.lst-kix_g65pgrp6l07-3>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-3,decimal) ". "}.lst-kix_g65pgrp6l07-4>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-4,lower-latin) ". "}.lst-kix_g65pgrp6l07-6>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-6,decimal) ". "}ol.lst-kix_51wdrdk2fq9o-3.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-3 0}.lst-kix_g65pgrp6l07-7>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-7,lower-latin) ". "}.lst-kix_g65pgrp6l07-8>li:before{content:"" counter(lst-ctn-kix_g65pgrp6l07-8,lower-roman) ". "}.lst-kix_90oyjc4qrwtp-7>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-7,lower-latin) ". "}.lst-kix_qwuufa9dxghx-3>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-3}.lst-kix_51wdrdk2fq9o-4>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-4}.lst-kix_90oyjc4qrwtp-8>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-8,lower-roman) ". "}ol.lst-kix_pqruqr21jqkv-4.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-4 0}.lst-kix_rcus5r4ke429-0>li{counter-increment:lst-ctn-kix_rcus5r4ke429-0}.lst-kix_epldur8jqq61-2>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-2,lower-roman) ". "}.lst-kix_epldur8jqq61-3>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-3,decimal) ". "}ol.lst-kix_pqruqr21jqkv-6.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-6 0}.lst-kix_epldur8jqq61-0>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-0,decimal) ". "}.lst-kix_epldur8jqq61-5>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-5,lower-roman) ". "}.lst-kix_90oyjc4qrwtp-1>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-1,lower-latin) ". "}ol.lst-kix_3qvy38k9i9l4-3.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-3 0}.lst-kix_epldur8jqq61-7>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-7,lower-latin) ". "}.lst-kix_51wdrdk2fq9o-7>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-7}ol.lst-kix_90oyjc4qrwtp-6.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-6 0}.lst-kix_90oyjc4qrwtp-5>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-5,lower-roman) ". "}ol.lst-kix_3qvy38k9i9l4-6.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-6 0}.lst-kix_g65pgrp6l07-0>li{counter-increment:lst-ctn-kix_g65pgrp6l07-0}.lst-kix_90oyjc4qrwtp-3>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-3,decimal) ". "}ol.lst-kix_li21gc20orp9-2.start{counter-reset:lst-ctn-kix_li21gc20orp9-2 0}ol.lst-kix_51wdrdk2fq9o-1.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-1 0}.lst-kix_bc3xa550q60r-8>li{counter-increment:lst-ctn-kix_bc3xa550q60r-8}ul.lst-kix_hedhuo5tdgo-8{list-style-type:none}ol.lst-kix_epldur8jqq61-1.start{counter-reset:lst-ctn-kix_epldur8jqq61-1 0}ul.lst-kix_hedhuo5tdgo-7{list-style-type:none}.lst-kix_li21gc20orp9-0>li{counter-increment:lst-ctn-kix_li21gc20orp9-0}.lst-kix_rcus5r4ke429-7>li{counter-increment:lst-ctn-kix_rcus5r4ke429-7}ol.lst-kix_epldur8jqq61-4.start{counter-reset:lst-ctn-kix_epldur8jqq61-4 0}ol.lst-kix_rcus5r4ke429-1.start{counter-reset:lst-ctn-kix_rcus5r4ke429-1 0}ol.lst-kix_2d4m39kn8zc9-2.start{counter-reset:lst-ctn-kix_2d4m39kn8zc9-2 0}.lst-kix_qwuufa9dxghx-6>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-6,decimal) ". "}.lst-kix_qwuufa9dxghx-8>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-8,lower-roman) ". "}ol.lst-kix_rcus5r4ke429-3.start{counter-reset:lst-ctn-kix_rcus5r4ke429-3 0}ol.lst-kix_bc3xa550q60r-6.start{counter-reset:lst-ctn-kix_bc3xa550q60r-6 0}.lst-kix_90oyjc4qrwtp-3>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-3}.lst-kix_51wdrdk2fq9o-4>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-4,lower-latin) ". "}.lst-kix_51wdrdk2fq9o-8>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-8,lower-roman) ". "}ol.lst-kix_li21gc20orp9-4.start{counter-reset:lst-ctn-kix_li21gc20orp9-4 0}.lst-kix_642xdqsnfsyh-6>li:before{content:"-  "}.lst-kix_51wdrdk2fq9o-6>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-6,decimal) ". "}.lst-kix_pqruqr21jqkv-5>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-5}.lst-kix_bc3xa550q60r-3>li{counter-increment:lst-ctn-kix_bc3xa550q60r-3}ul.lst-kix_hedhuo5tdgo-4{list-style-type:none}ol.lst-kix_li21gc20orp9-8{list-style-type:none}.lst-kix_qwuufa9dxghx-7>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-7}ul.lst-kix_hedhuo5tdgo-3{list-style-type:none}ol.lst-kix_li21gc20orp9-7{list-style-type:none}ol.lst-kix_epldur8jqq61-8{list-style-type:none}.lst-kix_642xdqsnfsyh-4>li:before{content:"-  "}ul.lst-kix_hedhuo5tdgo-6{list-style-type:none}ol.lst-kix_li21gc20orp9-6{list-style-type:none}ol.lst-kix_epldur8jqq61-7{list-style-type:none}ul.lst-kix_hedhuo5tdgo-5{list-style-type:none}ol.lst-kix_li21gc20orp9-5{list-style-type:none}ol.lst-kix_epldur8jqq61-6{list-style-type:none}ul.lst-kix_hedhuo5tdgo-0{list-style-type:none}ol.lst-kix_epldur8jqq61-5{list-style-type:none}ol.lst-kix_epldur8jqq61-4{list-style-type:none}ul.lst-kix_hedhuo5tdgo-2{list-style-type:none}.lst-kix_51wdrdk2fq9o-0>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-0,decimal) ". "}ol.lst-kix_epldur8jqq61-3{list-style-type:none}ul.lst-kix_hedhuo5tdgo-1{list-style-type:none}ol.lst-kix_epldur8jqq61-2{list-style-type:none}.lst-kix_rcus5r4ke429-8>li{counter-increment:lst-ctn-kix_rcus5r4ke429-8}ol.lst-kix_epldur8jqq61-1{list-style-type:none}.lst-kix_642xdqsnfsyh-0>li:before{content:"-  "}ol.lst-kix_epldur8jqq61-0{list-style-type:none}.lst-kix_51wdrdk2fq9o-0>li{counter-increment:lst-ctn-kix_51wdrdk2fq9o-0}.lst-kix_642xdqsnfsyh-2>li:before{content:"-  "}.lst-kix_52z049lpujuv-5>li{counter-increment:lst-ctn-kix_52z049lpujuv-5}.lst-kix_51wdrdk2fq9o-2>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-2,lower-roman) ". "}ol.lst-kix_g65pgrp6l07-0.start{counter-reset:lst-ctn-kix_g65pgrp6l07-0 0}.lst-kix_3qvy38k9i9l4-5>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-5,lower-roman) ". "}ol.lst-kix_90oyjc4qrwtp-8.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-8 0}ol.lst-kix_bc3xa550q60r-5.start{counter-reset:lst-ctn-kix_bc3xa550q60r-5 0}ol.lst-kix_li21gc20orp9-0{list-style-type:none}.lst-kix_3qvy38k9i9l4-7>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-7,lower-latin) ". "}ol.lst-kix_li21gc20orp9-4{list-style-type:none}ol.lst-kix_li21gc20orp9-3{list-style-type:none}ol.lst-kix_li21gc20orp9-2{list-style-type:none}ol.lst-kix_li21gc20orp9-1{list-style-type:none}.lst-kix_xf4ne1qm8zsq-2>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-2}ol.lst-kix_qwuufa9dxghx-8.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-8 0}ol.lst-kix_52z049lpujuv-6.start{counter-reset:lst-ctn-kix_52z049lpujuv-6 0}.lst-kix_90oyjc4qrwtp-6>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-6}.lst-kix_epldur8jqq61-3>li{counter-increment:lst-ctn-kix_epldur8jqq61-3}ol.lst-kix_qwuufa9dxghx-3.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-3 0}.lst-kix_xf4ne1qm8zsq-3>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-3,decimal) ". "}.lst-kix_52z049lpujuv-2>li{counter-increment:lst-ctn-kix_52z049lpujuv-2}.lst-kix_xf4ne1qm8zsq-5>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-5}ol.lst-kix_52z049lpujuv-8{list-style-type:none}.lst-kix_52z049lpujuv-0>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-0,decimal) ". "}ol.lst-kix_52z049lpujuv-7{list-style-type:none}.lst-kix_xf4ne1qm8zsq-4>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-4,lower-latin) ". "}ol.lst-kix_52z049lpujuv-6{list-style-type:none}ol.lst-kix_52z049lpujuv-5{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-6{list-style-type:none}ol.lst-kix_52z049lpujuv-4{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-5{list-style-type:none}.lst-kix_52z049lpujuv-1>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-1,lower-latin) ". "}ol.lst-kix_52z049lpujuv-3{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-4{list-style-type:none}ol.lst-kix_52z049lpujuv-2{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-3{list-style-type:none}.lst-kix_pqruqr21jqkv-2>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-2}ol.lst-kix_52z049lpujuv-1{list-style-type:none}.lst-kix_xf4ne1qm8zsq-7>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-7,lower-latin) ". "}ol.lst-kix_52z049lpujuv-0{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-8{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-7{list-style-type:none}.lst-kix_bc3xa550q60r-1>li{counter-increment:lst-ctn-kix_bc3xa550q60r-1}.lst-kix_52z049lpujuv-5>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-5,lower-roman) ". "}ol.lst-kix_51wdrdk2fq9o-5.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-5 0}.lst-kix_g65pgrp6l07-6>li{counter-increment:lst-ctn-kix_g65pgrp6l07-6}ol.lst-kix_3qvy38k9i9l4-2{list-style-type:none}ol.lst-kix_3qvy38k9i9l4-1{list-style-type:none}.lst-kix_52z049lpujuv-4>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-4,lower-latin) ". "}ol.lst-kix_3qvy38k9i9l4-0{list-style-type:none}ol.lst-kix_rcus5r4ke429-5.start{counter-reset:lst-ctn-kix_rcus5r4ke429-5 0}.lst-kix_2d4m39kn8zc9-4>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-4}.lst-kix_xf4ne1qm8zsq-8>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-8,lower-roman) ". "}.lst-kix_xf4ne1qm8zsq-3>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-3}ol.lst-kix_xf4ne1qm8zsq-8.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-8 0}.lst-kix_pqruqr21jqkv-4>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-4}.lst-kix_52z049lpujuv-4>li{counter-increment:lst-ctn-kix_52z049lpujuv-4}.lst-kix_90oyjc4qrwtp-4>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-4}.lst-kix_52z049lpujuv-8>li:before{content:"" counter(lst-ctn-kix_52z049lpujuv-8,lower-roman) ". "}.lst-kix_rcus5r4ke429-1>li{counter-increment:lst-ctn-kix_rcus5r4ke429-1}ol.lst-kix_li21gc20orp9-6.start{counter-reset:lst-ctn-kix_li21gc20orp9-6 0}ol.lst-kix_pqruqr21jqkv-2.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-2 0}.lst-kix_642xdqsnfsyh-8>li:before{content:"-  "}.lst-kix_g65pgrp6l07-8>li{counter-increment:lst-ctn-kix_g65pgrp6l07-8}ol.lst-kix_xf4ne1qm8zsq-2.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-2 0}ol.lst-kix_3qvy38k9i9l4-5.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-5 0}ol.lst-kix_xf4ne1qm8zsq-7.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-7 0}.lst-kix_2d4m39kn8zc9-2>li{counter-increment:lst-ctn-kix_2d4m39kn8zc9-2}.lst-kix_li21gc20orp9-4>li{counter-increment:lst-ctn-kix_li21gc20orp9-4}.lst-kix_pqruqr21jqkv-0>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-0}.lst-kix_xf4ne1qm8zsq-7>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-7}ol.lst-kix_li21gc20orp9-7.start{counter-reset:lst-ctn-kix_li21gc20orp9-7 0}ol.lst-kix_90oyjc4qrwtp-4.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-4 0}ol.lst-kix_xf4ne1qm8zsq-1.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-1 0}.lst-kix_qwuufa9dxghx-0>li{counter-increment:lst-ctn-kix_qwuufa9dxghx-0}.lst-kix_rcus5r4ke429-6>li{counter-increment:lst-ctn-kix_rcus5r4ke429-6}ol.lst-kix_xf4ne1qm8zsq-8{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-7{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-6{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-5{list-style-type:none}.lst-kix_52z049lpujuv-0>li{counter-increment:lst-ctn-kix_52z049lpujuv-0}ol.lst-kix_xf4ne1qm8zsq-4{list-style-type:none}ol.lst-kix_90oyjc4qrwtp-5.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-5 0}ol.lst-kix_xf4ne1qm8zsq-3{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-2{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-1{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-0{list-style-type:none}ol.lst-kix_rcus5r4ke429-6.start{counter-reset:lst-ctn-kix_rcus5r4ke429-6 0}.lst-kix_epldur8jqq61-7>li{counter-increment:lst-ctn-kix_epldur8jqq61-7}.lst-kix_90oyjc4qrwtp-8>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-8}.lst-kix_xf4ne1qm8zsq-0>li:before{content:"" counter(lst-ctn-kix_xf4ne1qm8zsq-0,decimal) ". "}.lst-kix_rcus5r4ke429-7>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-7,lower-latin) ". "}.lst-kix_90oyjc4qrwtp-0>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-0,decimal) ". "}.lst-kix_bc3xa550q60r-6>li{counter-increment:lst-ctn-kix_bc3xa550q60r-6}.lst-kix_epldur8jqq61-1>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-1,lower-latin) ". "}.lst-kix_epldur8jqq61-4>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-4,lower-latin) ". "}.lst-kix_rcus5r4ke429-3>li:before{content:"" counter(lst-ctn-kix_rcus5r4ke429-3,decimal) ". "}.lst-kix_3qvy38k9i9l4-2>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-2}ol.lst-kix_g65pgrp6l07-1{list-style-type:none}.lst-kix_90oyjc4qrwtp-4>li:before{content:"" counter(lst-ctn-kix_90oyjc4qrwtp-4,lower-latin) ". "}ol.lst-kix_g65pgrp6l07-2{list-style-type:none}.lst-kix_epldur8jqq61-8>li:before{content:"" counter(lst-ctn-kix_epldur8jqq61-8,lower-roman) ". "}ol.lst-kix_g65pgrp6l07-0{list-style-type:none}ol.lst-kix_g65pgrp6l07-5{list-style-type:none}ol.lst-kix_g65pgrp6l07-6{list-style-type:none}ol.lst-kix_xf4ne1qm8zsq-6.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-6 0}ol.lst-kix_g65pgrp6l07-3{list-style-type:none}ol.lst-kix_g65pgrp6l07-4{list-style-type:none}ol.lst-kix_g65pgrp6l07-7{list-style-type:none}ol.lst-kix_g65pgrp6l07-8{list-style-type:none}ol.lst-kix_pqruqr21jqkv-3.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-3 0}ol.lst-kix_qwuufa9dxghx-4.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-4 0}ol.lst-kix_90oyjc4qrwtp-3.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-3 0}.lst-kix_pqruqr21jqkv-7>li{counter-increment:lst-ctn-kix_pqruqr21jqkv-7}.lst-kix_52z049lpujuv-7>li{counter-increment:lst-ctn-kix_52z049lpujuv-7}.lst-kix_xf4ne1qm8zsq-0>li{counter-increment:lst-ctn-kix_xf4ne1qm8zsq-0}ol.lst-kix_rcus5r4ke429-7.start{counter-reset:lst-ctn-kix_rcus5r4ke429-7 0}.lst-kix_rcus5r4ke429-4>li{counter-increment:lst-ctn-kix_rcus5r4ke429-4}ol.lst-kix_51wdrdk2fq9o-4.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-4 0}ol.lst-kix_3qvy38k9i9l4-8.start{counter-reset:lst-ctn-kix_3qvy38k9i9l4-8 0}.lst-kix_90oyjc4qrwtp-1>li{counter-increment:lst-ctn-kix_90oyjc4qrwtp-1}.lst-kix_bc3xa550q60r-0>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-0,decimal) ". "}.lst-kix_hedhuo5tdgo-0>li:before{content:"-  "}.lst-kix_3qvy38k9i9l4-0>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-0}ol.lst-kix_90oyjc4qrwtp-0.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-0 0}.lst-kix_epldur8jqq61-0>li{counter-increment:lst-ctn-kix_epldur8jqq61-0}.lst-kix_qwuufa9dxghx-7>li:before{content:"" counter(lst-ctn-kix_qwuufa9dxghx-7,lower-latin) ". "}.lst-kix_bc3xa550q60r-8>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-8,lower-roman) ". "}ol.lst-kix_pqruqr21jqkv-1.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-1 0}ol.lst-kix_xf4ne1qm8zsq-4.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-4 0}.lst-kix_51wdrdk2fq9o-5>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-5,lower-roman) ". "}.lst-kix_p5y3sb60oxh6-4>li:before{content:"-  "}.lst-kix_642xdqsnfsyh-5>li:before{content:"-  "}ol.lst-kix_51wdrdk2fq9o-7.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-7 0}ol.lst-kix_qwuufa9dxghx-1.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-1 0}.lst-kix_li21gc20orp9-1>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-1,lower-latin) ". "}.lst-kix_epldur8jqq61-5>li{counter-increment:lst-ctn-kix_epldur8jqq61-5}ol.lst-kix_rcus5r4ke429-8{list-style-type:none}ol.lst-kix_rcus5r4ke429-6{list-style-type:none}ol.lst-kix_rcus5r4ke429-7{list-style-type:none}ol.lst-kix_rcus5r4ke429-4{list-style-type:none}.lst-kix_pqruqr21jqkv-1>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-1,lower-latin) ". "}ol.lst-kix_90oyjc4qrwtp-1.start{counter-reset:lst-ctn-kix_90oyjc4qrwtp-1 0}ol.lst-kix_rcus5r4ke429-5{list-style-type:none}ol.lst-kix_rcus5r4ke429-2{list-style-type:none}.lst-kix_642xdqsnfsyh-1>li:before{content:"-  "}.lst-kix_li21gc20orp9-5>li:before{content:"" counter(lst-ctn-kix_li21gc20orp9-5,lower-roman) ". "}ol.lst-kix_rcus5r4ke429-3{list-style-type:none}.lst-kix_bc3xa550q60r-4>li:before{content:"" counter(lst-ctn-kix_bc3xa550q60r-4,lower-latin) ". "}ol.lst-kix_rcus5r4ke429-0{list-style-type:none}.lst-kix_51wdrdk2fq9o-1>li:before{content:"" counter(lst-ctn-kix_51wdrdk2fq9o-1,lower-latin) ". "}ol.lst-kix_rcus5r4ke429-1{list-style-type:none}.lst-kix_p5y3sb60oxh6-0>li:before{content:"-  "}.lst-kix_g65pgrp6l07-3>li{counter-increment:lst-ctn-kix_g65pgrp6l07-3}.lst-kix_pqruqr21jqkv-5>li:before{content:"" counter(lst-ctn-kix_pqruqr21jqkv-5,lower-roman) ". "}ol.lst-kix_51wdrdk2fq9o-6.start{counter-reset:lst-ctn-kix_51wdrdk2fq9o-6 0}.lst-kix_li21gc20orp9-2>li{counter-increment:lst-ctn-kix_li21gc20orp9-2}ol.lst-kix_pqruqr21jqkv-0.start{counter-reset:lst-ctn-kix_pqruqr21jqkv-0 0}ol.lst-kix_xf4ne1qm8zsq-3.start{counter-reset:lst-ctn-kix_xf4ne1qm8zsq-3 0}.lst-kix_3qvy38k9i9l4-6>li:before{content:"" counter(lst-ctn-kix_3qvy38k9i9l4-6,decimal) ". "}ol.lst-kix_qwuufa9dxghx-2.start{counter-reset:lst-ctn-kix_qwuufa9dxghx-2 0}li.li-bullet-0:before{margin-left:-18pt;white-space:nowrap;display:inline-block;min-width:18pt}.lst-kix_3qvy38k9i9l4-7>li{counter-increment:lst-ctn-kix_3qvy38k9i9l4-7}.lst-kix_p5y3sb60oxh6-8>li:before{content:"-  "}ol{margin:0;padding:0}table td,table th{padding:0}.c1{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:0pt;border-right-width:0pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:0pt;border-top-style:solid;border-left-style:solid;border-bottom-width:0pt;width:47pt;border-top-color:#000000;border-bottom-style:solid}.c40{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:0pt;border-right-width:0pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:0pt;border-top-style:solid;border-left-style:solid;border-bottom-width:0pt;width:247.5pt;border-top-color:#000000;border-bottom-style:solid}.c27{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:0pt;border-right-width:0pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:0pt;border-top-style:solid;border-left-style:solid;border-bottom-width:0pt;width:99pt;border-top-color:#000000;border-bottom-style:solid}.c8{border-right-style:solid;padding:5pt 5pt 5pt 5pt;border-bottom-color:#000000;border-top-width:0pt;border-right-width:0pt;border-left-color:#000000;vertical-align:top;border-right-color:#000000;border-left-width:0pt;border-top-style:solid;border-left-style:solid;border-bottom-width:0pt;width:57.8pt;border-top-color:#000000;border-bottom-style:solid}.c3{margin-left:36pt;padding-top:0pt;padding-left:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c22{padding-top:20pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left;height:20pt}.c44{padding-top:18pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c30{padding-top:20pt;padding-bottom:6pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c18{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:20pt;font-family:"Arial";font-style:normal}.c2{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left;height:11pt}.c9{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:9pt;font-family:"Arial";font-style:normal}.c10{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:normal}.c39{padding-top:0pt;padding-bottom:16pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c17{color:#e69138;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:11pt;font-family:"Arial";font-style:italic}.c34{padding-top:0pt;padding-bottom:3pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c35{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:16pt;font-family:"Arial";font-style:normal}.c33{color:#666666;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:15pt;font-family:"Arial";font-style:normal}.c38{padding-top:16pt;padding-bottom:4pt;line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.c6{color:#434343;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:14pt;font-family:"Arial";font-style:normal}.c4{padding-top:0pt;padding-bottom:0pt;line-height:1.15;orphans:2;widows:2;text-align:left}.c46{color:#000000;text-decoration:none;vertical-align:baseline;font-size:26pt;font-style:normal}.c14{color:#000000;text-decoration:none;vertical-align:baseline;font-size:6pt;font-style:normal}.c21{color:#38761d;text-decoration:none;vertical-align:baseline;font-size:11pt;font-style:normal}.c25{color:#000000;text-decoration:none;vertical-align:baseline;font-size:9pt;font-style:normal}.c16{color:#000000;text-decoration:none;vertical-align:baseline;font-size:11pt;font-style:normal}.c13{color:#6aa84f;text-decoration:none;vertical-align:baseline;font-size:11pt;font-style:normal}.c32{padding-top:3pt;padding-bottom:0pt;line-height:1.0;text-align:left}.c36{color:#000000;text-decoration:none;vertical-align:baseline;font-size:11pt}.c42{border-spacing:0;border-collapse:collapse;margin-right:auto}.c12{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c41{background-color:#ffffff;max-width:468pt;padding:72pt 72pt 72pt 72pt}.c19{color:inherit;text-decoration:inherit}.c7{padding:0;margin:0}.c0{font-weight:400;font-family:"Courier New"}.c23{font-weight:700;font-family:"Arial"}.c43{font-weight:400;font-family:"Arial"}.c24{color:#38761d}.c26{margin-left:36pt}.c47{font-family:"Courier New"}.c20{margin-left:108pt}.c28{font-weight:700}.c15{height:34.7pt}.c31{font-size:6pt}.c45{margin-left:18pt}.c37{background-color:#fff2cc}.c5{margin-left:72pt}.c11{padding-left:0pt}.c29{font-style:italic}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c41 doc-content"><p class="c34 title" id="h.gzhwrkcii2d9"><span class="c43 c46">Building a Peptide Atlas on Liverpool HPC</span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.19u56f9ykxms">Introduction&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1</a></span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.kryms3l06ooq">1. Making a fasta search database&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1</a></span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.dif9jmngzc9k">2. Choosing and retrieving peptide data&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;3</a></span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.ml8nlf6lm0za">3. Processing peptide data&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;6</a></span></p><p class="c32 c45"><span class="c10"><a class="c19" href="#h.o7lqkvhrlhpa">3a. Search tools&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;6</a></span></p><p class="c32 c45"><span class="c10"><a class="c19" href="#h.onohkzfg7ske">3b. Instrument parameters&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;6</a></span></p><p class="c32 c45"><span class="c10"><a class="c19" href="#h.1gl7rvyauw8k">3c. QC of data&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;8</a></span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.imhqa9nobeaw">4. Combining and collapsing data&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;9</a></span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.wgxmc6v1wcpe">5. Visualising peptides on a genome&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;9</a></span></p><p class="c32"><span class="c16 c23"><a class="c19" href="#h.oxvtznduwpe0">Running the pipeline&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;10</a></span></p><p class="c2"><span class="c10"></span></p><h1 class="c30" id="h.19u56f9ykxms"><span class="c18">Introduction</span></h1><p class="c4"><span class="c10">For this task, we want to get peptide-level support for genes within a genome, sometimes called proteogenomics. We source mass spectrometry (MS) data from ProteomeXchange, and its underlying repositories - PRIDE, MassIVE, IProx etc. We then define a search database (see below), and run the Trans Proteomic Pipeline on our linux HPC, to search MS raw data against the database, generate statistics etc. The data is then passed to Eric Deutsch at PeptideAtlas, who loads it into their database, and does a few extra steps of post-processing. We then get data out in TSV format. </span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c28">Several steps of this guide have been converted into a set of scripts and a pipeline (process-pxd.sh) that runs them all together. </span><span>The github for all these files can be found here: </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://github.com/VibrantStarling/make-protein-atlas&amp;sa=D&amp;source=editors&amp;ust=1751561387984595&amp;usg=AOvVaw1DIIhXCCi2LX2tnWO_KJby">https://github.com/VibrantStarling/make-protein-atlas</a></span><span>. You can also find further instruction on how to use this repository at the bottom of this document, </span><span class="c12"><a class="c19" href="#h.oxvtznduwpe0">here</a></span><span class="c10">.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">But first, a step by step explanation of what we&rsquo;re doing&hellip;</span></p><hr style="page-break-before:always;display:none;"><h1 class="c30" id="h.kryms3l06ooq"><span class="c18">1. Making a fasta search database</span></h1><p class="c4"><span class="c10">The fasta database is a compilation of all the relevant proteins that you want to be able to find in your peptide data, including common contaminants and reverse reads. The makeup of a database will depend on what you are looking for, but generally you want a balance of size and specificity. A large database with an excess of irrelevant reads increases computation time without improving statistical power. Conversely, a small database with highly specific reads could be very efficient, but give misleading results. </span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">There is no hard and fast rule for what to include in a database. Try out a few variations, or talk to your supervisor and colleagues as to what seem reasonable</span></p><p class="c2"><span class="c10"></span></p><h2 class="c44" id="h.bamfgo1hjczy"><span class="c35">Starting out</span></h2><p class="c4"><span>Basic principles - when performing proteogenomics, we generally start from an assumption that the &ldquo;official&rdquo; gene models are not completely correct. We want to find peptides that map onto the proteins from those gene models (the &ldquo;official&rdquo; proteome), but also find peptides that could &nbsp;map to other gene models. These gene models may have been predicted by pipelines in the past - like &ldquo;Ensembl </span><span class="c29">ab initio</span><span>&rdquo; models, by running </span><span class="c29">de novo </span><span>predictions e.g. </span><span>with BRAKER3 / FunAnnotate</span><span>&nbsp;(instructions on how to do this: </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://github.com/VibrantStarling/genomics-from-the-ground-up&amp;sa=D&amp;source=editors&amp;ust=1751561387987034&amp;usg=AOvVaw2SXshxb82kdZCvCytlUykw">https://github.com/VibrantStarling/genomics-from-the-ground-up</a></span><span class="c10">&nbsp;), or by assembling transcriptomics data into plausible transcripts. For example, if long read (ISO-SEQ) transcripts exist, these can be translated in three frames, taking all open reading frames that run from an ATG to a stop codon. </span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span>It is generally not advisable to perform a six-frame genome translation. This produces a very large search database (losing statistical power), and does not predict intron-spanning peptides (ISPs). ISPs are some of the most useful datatypes we can get from proteomics.</span></p><p class="c2"><span class="c16 c23"></span></p><p class="c2 c26"><span class="c10"></span></p><h2 class="c44" id="h.7xrsrzy6atl"><span class="c35">Making your own proteome fasta</span></h2><p class="c4"><span class="c10">There are several ways to compile fasta sequences for a search database. The end aim is to have a range of protein sequences in fasta format that represent genes you want to find within the mass spec protein data.</span></p><h3 class="c38" id="h.l38isfhw7z5j"><span>1. </span><span class="c6">Establish a base from:</span></h3><p class="c4"><span class="c16 c23">Preexisting proteome data</span></p><ol class="c7 lst-kix_52z049lpujuv-0 start" start="1"><li class="c3 li-bullet-0"><span>Find your reference proteome on Uniprot, VEuPathDB, NCBI or wherever has the best annotation for your species, e.g. </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://www.uniprot.org/uniprotkb?query%3Dproteome:UP000002311&amp;sa=D&amp;source=editors&amp;ust=1751561387989245&amp;usg=AOvVaw0TRgY5AHw9VF0mm_Njrp1H">https://www.uniprot.org/uniprotkb?query=proteome:UP000002311</a></span><span class="c10">&nbsp;</span></li><li class="c3 li-bullet-0"><span class="c10">Download the proteome in fasta format, and make sure it is amino acid sequence</span></li></ol><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c16 c23">Amnio acid fasta files</span></p><ol class="c7 lst-kix_pqruqr21jqkv-0 start" start="1"><li class="c3 li-bullet-0"><span class="c10">Select your fasta files with your desired genes</span></li><li class="c3 li-bullet-0"><span>If you have multiple files, merge all protein fasta files into one, e.g. using cat *.fasta &gt; ../final_database/full_species_merged.fasta. </span><span class="c36 c43 c29">It does not matter if you have repeat sequences.</span></li></ol><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c16 c23">gff/gtf files</span></p><ol class="c7 lst-kix_rcus5r4ke429-0 start" start="1"><li class="c3 li-bullet-0"><span class="c10">Retrieve the gff or gtf of the sets of proteins you are interested in searching (or the amino acid fasta file with an associated gff). You might have one or several.</span></li><li class="c3 li-bullet-0"><span class="c10">Download the nucleotide genome fasta for the gff/gtf</span></li><li class="c3 li-bullet-0"><span>Use something like </span><span class="c0 c24">gffread </span><span>or </span><span class="c0 c24">AGAT </span><span>to extract the </span><span class="c29">protein</span><span>&nbsp;amino acid sequence </span><span class="c10">from the gff. </span></li></ol><ol class="c7 lst-kix_rcus5r4ke429-2 start" start="1"><li class="c4 c20 c11 li-bullet-0"><span class="c13 c0">gffread -y CDS_aa.faa -g genome.fa input.gff</span></li><li class="c4 c20 c11 li-bullet-0"><span class="c0 c13">agat_sp_extract_sequences.pl -g infile.gff -f infile.fasta -t cds -p</span></li></ol><ol class="c7 lst-kix_rcus5r4ke429-0" start="4"><li class="c3 li-bullet-0"><span class="c10">Then merge all protein fasta files into one, e.g. using cat *.fasta &gt; ../final_database/full_species_merged.fasta</span></li></ol><p class="c2"><span class="c10"></span></p><h3 class="c38" id="h.z0t3li9ttiag"><span>2. </span><span class="c6">Add contaminants and decoys </span></h3><p class="c4"><span>With </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://fragpipe.nesvilab.org/docs/tutorial_fragpipe.html%23specify-a-protein-sequence-database&amp;sa=D&amp;source=editors&amp;ust=1751561387992035&amp;usg=AOvVaw0t0NkTvVqvuMQjDvYA4YTo">FragPipe</a></span><span class="c10">&nbsp;- this quickly and simply adds both contaminants and decoys in one and produces everything as one fasta. Make a note of your decoy prefix. Fragpipe can be downloaded and run as a GUI program as depicted below, or in commandline.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c16 c23">You can pass a fasta file (combined from the official proteins and your extra annotations) to fragpipe like this:</span></p><p class="c4"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 284.00px;"><img alt="" src="images/image3.png" style="width: 624.00px; height: 284.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2 c5"><span class="c10"></span></p><p class="c2"><span class="c16 c23"></span></p><p class="c2"><span class="c10"></span></p><h1 class="c30" id="h.dif9jmngzc9k"><span class="c18">2. Choosing and retrieving peptide data</span></h1><p class="c4"><span class="c10">When retrieving your datasets, make sure you target your species of interest. For pathogens in particular, you want to ensure data is NOT from host infected proteomes, although it can be co-cultured datasets if this is common for a given species. For these cases, you should include the host proteome (e.g. human) in the search database.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">For general global proteomics, you want to avoid datasets enriched for post translational modifications (PTMs) like phosphorylation, glycosylation or acetylation etc. You also want to avoid interactome and TurboID sets which usually only identify a few proteins. Ideally, source articles for datasets would mention that they have identified &gt;2K - 3K proteins, or a minimum say of &gt;1K. </span></p><p class="c2"><span class="c10"></span></p><p class="c39 subtitle" id="h.wy8hznykkv0d"><span class="c33">Choosing datasets</span></p><ol class="c7 lst-kix_qwuufa9dxghx-0 start" start="1"><li class="c3 li-bullet-0"><span>Go to </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://proteomecentral.proteomexchange.org/&amp;sa=D&amp;source=editors&amp;ust=1751561387994891&amp;usg=AOvVaw1lkKfuKtVn4eLP6yhUmqTL">https://proteomecentral.proteomexchange.org/</a></span><span class="c10">&nbsp;and search for your species. </span></li><li class="c3 li-bullet-0"><span class="c10">Download the results as a TSV, then move in to google sheets to annotate them.</span></li><li class="c3 li-bullet-0"><span class="c10">Add a column for:</span></li></ol><ol class="c7 lst-kix_qwuufa9dxghx-1 start" start="1"><li class="c4 c5 c11 li-bullet-0"><span class="c28">Priority &nbsp;</span><span class="c10">- for scoring how likely you are to use them (0,1,2 -&gt; No, Maybe, Yes)</span></li><li class="c4 c5 c11 li-bullet-0"><span class="c28">Notes </span><span class="c10">- keep track of reasons for your scoring, particularly Maybes</span></li><li class="c4 c5 c11 li-bullet-0"><span class="c28">Parameters </span><span>- track what label type was used (TMT, iTRAQ, label-free, </span><span class="c29">etc</span><span class="c10">.). This will determine what parameters we will use for MS-Fragger searches</span></li><li class="c4 c5 c11 li-bullet-0"><span class="c28">Instrument </span><span class="c10">- track what instrument was used for the dataset </span></li></ol><ol class="c7 lst-kix_qwuufa9dxghx-2 start" start="1"><li class="c4 c11 c20 li-bullet-0"><span class="c10">Favour Thermo instruments. Orbitrap Fusion, Orbitrap Lumos, Orbitrap Astral, Q-Exactive, Q-Exactive HF are best. Maybe include LTQ Orbitrap if paper reports it is a big data sets e.g. identifying &gt;2000 proteins). LTQ Orbitrap have low-res fragment ions, and need this setting change in search parameters</span></li><li class="c4 c20 c11 li-bullet-0"><span class="c10">Exclude Waters (SYNAPT) instruments, and usually exclude Bruker (timsTOF) and SCIEX (TripleTOF), unless you are struggling for data.</span></li></ol><ol class="c7 lst-kix_qwuufa9dxghx-1" start="5"><li class="c4 c5 c11 li-bullet-0"><span class="c28">Data dependent acquisition (DDA) </span><span>-</span><span class="c28">&nbsp;</span><span>If paper or submission mentions Data </span><span>independent </span><span class="c10">acquisition (DIA), we exclude. </span></li></ol><ol class="c7 lst-kix_qwuufa9dxghx-2 start" start="1"><li class="c4 c20 c11 li-bullet-0"><span class="c10">Markers of DIA include &nbsp;(aka. SWATH) include using software DIA-NN, MaxDIA, Spectronaut or Skyline. </span></li><li class="c4 c20 c11 li-bullet-0"><span class="c10">Markers of DDA including using MaxQuant, ProteomeDiscoverer, FragPipe&hellip;</span></li></ol><ol class="c7 lst-kix_qwuufa9dxghx-1" start="6"><li class="c4 c5 c11 li-bullet-0"><span class="c28">Source database. </span><span class="c10">It is easier to get fast download from PRIDE direct to the Linux cluster. If large datasets are in other databases, we can get them via a more manual process e.g. using FileZilla for FTP transfer to a local PC, then copying them to the cluster, but this is slow and annoying!</span></li></ol><ol class="c7 lst-kix_qwuufa9dxghx-0" start="4"><li class="c3 li-bullet-0"><span class="c10">Go through the sheet and find the datasets you want to use. </span></li><li class="c3 li-bullet-0"><span class="c10">Often the link to a source publication is missing. If you google search for the PXD identifier, you can often find this embedded in a paper. Keep track of this (add manually to the publication column). We can inform PRIDE support to add the publication link to the dataset.</span></li></ol><p class="c2"><span class="c10"></span></p><p class="c39 subtitle" id="h.h6atpljk2x2z"><span class="c33">Downloading datasets</span></p><p class="c4"><span class="c28 c37">NOTE - </span><span class="c10 c37">this step is included in the pipeline at the end of this document and can handle multiple PXDs at once. This section breaks down what you would need to do to do it yourself for one dataset.</span></p><p class="c2"><span class="c10 c37"></span></p><ol class="c7 lst-kix_bc3xa550q60r-0 start" start="1"><li class="c3 li-bullet-0"><span>On the login/head node on HPC make a directory for your species in &nbsp;</span><span class="c0">/mnt/hc-storage/groups/peptideatlas_builds/. </span><span class="c10">Avoid spaces in your directory name, call it something informative like Toxoplasma_gondii_ME49, ToxoplasmaGondiiME49, or TgondiiME49</span></li><li class="c3 li-bullet-0"><span class="c10">Make a subfolder per PXD accession number (ProteomeXchange ID)</span></li></ol><p class="c4 c5"><span class="c21 c0">mkdir PXD0123456</span></p><ol class="c7 lst-kix_bc3xa550q60r-0" start="3"><li class="c3 li-bullet-0"><span class="c10">Navigate to PXD subfolder, then do the following for datasets in PRIDE only</span></li></ol><p class="c4 c5"><span class="c21 c0">module load aspera</span></p><p class="c4 c5"><span class="c21 c0">module load pgbtools</span></p><p class="c4 c5"><span class="c21 c0">FetchPride2 PXD0123456</span></p><ol class="c7 lst-kix_bc3xa550q60r-0" start="4"><li class="c3 li-bullet-0"><span>You will either need to leave the terminal window open or run the code with </span><span class="c0">nohup </span><span class="c10">to make it run in the background without risking the terminal timing out. For instance:</span></li></ol><p class="c4 c5"><span class="c21 c0">nohup FetchPride2 PXD0123456</span></p><ol class="c7 lst-kix_bc3xa550q60r-0" start="5"><li class="c3 li-bullet-0"><span>Check that your downloads have worked</span></li></ol><hr style="page-break-before:always;display:none;"><h1 class="c22" id="h.9yhzod9gca67"><span class="c18"></span></h1><h1 class="c30" id="h.ml8nlf6lm0za"><span>3. </span><span class="c18">Processing peptide data</span></h1><p class="c4"><span>The code for these steps can be found in this slurm script: </span><span class="c0 c24">TPP-fragger-wholedir-postprocess-GENERAL.sh</span></p><h2 class="c44" id="h.o7lqkvhrlhpa"><span>3a. </span><span class="c35">Search tools</span></h2><p class="c4"><span class="c10">A typical pipeline has these steps, all run with SLURM:</span></p><p class="c2"><span class="c10"></span></p><ol class="c7 lst-kix_g65pgrp6l07-0 start" start="1"><li class="c3 li-bullet-0"><span class="c10">Convert raw to mzML with msconvert</span></li><li class="c3 li-bullet-0"><span class="c10">Run search with Comet or MS-Fragger - for peptide atlas tasks Andy favours MS-fragger for higher speed</span></li></ol><ol class="c7 lst-kix_g65pgrp6l07-1 start" start="1"><li class="c4 c5 c11 li-bullet-0"><span class="c16 c23">Absolute key step is setting the variable modifications and instrument tolerances correctly.</span></li><li class="c4 c5 c11 li-bullet-0"><span class="c10">There are pros and cons to using array jobs here, Andy prefers a single slurm script that doesn&rsquo;t use array jobs but locks 80 threads (one machine) per PXD from start to finish (if cluster is quiet, this is perfect, if busy, it is less ideal - potentially hogging resources in some steps). A single job is default in the pipeline script too.</span></li></ol><ol class="c7 lst-kix_g65pgrp6l07-0" start="3"><li class="c3 li-bullet-0"><span class="c10">Run TPP tools - PeptideProphet (xinteract) and IProphet (InterProphetParser)</span></li><li class="c3 li-bullet-0"><span class="c10">Convert pepXML to tsv and do hacky post-processing to check everything has worked (Andy&rsquo;s python scripts). </span></li></ol><ol class="c7 lst-kix_g65pgrp6l07-1 start" start="1"><li class="c4 c5 c11 li-bullet-0"><span class="c16 c23">I would like to re-engineer these final steps to produce a clean &ldquo;build&rdquo; for VEuPathDB, including converting peptides to BED or BAM format for genome display, and sorting peptides that are only present in some annotation sets but not others. This is on the TO DO list for someone in the team.</span></li></ol><h2 class="c44" id="h.onohkzfg7ske"><span>3b. </span><span class="c35">Instrument parameters</span></h2><p class="c4"><span>MS-fragger requires a file that contains specific parameters for each type of data (label-free, TMT, TMTpro, etc.). In some cases this might be changing the mass of cysteine or lysine to account for different labeling. A summary of MS-fragger parameters can be found </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://github-wiki-see.page/m/Nesvilab/MSFragger/wiki/Setting-the-Parameters&amp;sa=D&amp;source=editors&amp;ust=1751561388003617&amp;usg=AOvVaw03DX-JUPszTCkTBu8Ebd5_">here</a></span><span class="c10">. </span></p><p class="c4"><span class="c10">The most important settings are getting the variable and fixed modifications correct. If these are incorrect, we will not identify many peptides. Assuming we are not searching for enriched PTMs, then we have to focus on learning the quantification reagents (if any) used.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span>We have several </span><span class="c0 c24">.param</span><span>&nbsp;files with set parameters for the most common label types we come across.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span>The main thing you need to remember to change across all .param files &nbsp;are the </span><span class="c0 c24">num_threads </span><span>and </span><span class="c0 c24">database_name </span><span class="c10">options. This is done automatically in the pipelines, but is something you should bear in mind.</span></p><p class="c2"><span class="c16 c23"></span></p><p class="c4"><span class="c16 c23">Other very important parts of the param file:</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c0 c24">variable_mod_01 .. 16</span><span class="c0">&nbsp;</span><span>Sets variable modifications (variable_mod_01 to variable_mod_16). Space separated values with 1st value being the modification mass and the second being the residues (specified consecutively as a string) it modifies. You can find information on the syntax </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://github-wiki-see.page/m/Nesvilab/MSFragger/wiki/Setting-the-Parameters&amp;sa=D&amp;source=editors&amp;ust=1751561388005322&amp;usg=AOvVaw2upAS-OnMZ5JUShDBHPG0Z">here</a></span><span class="c10">. Here are some examples from one of the .param files:</span></p><p class="c2"><span class="c10"></span></p><p class="c4 c26"><span class="c0 c24">variable_mod_01 = 15.994900 M 3</span><span class="c0">&nbsp;</span><span class="c10">(methionine oxidation)</span></p><p class="c4 c26"><span class="c0 c24">variable_mod_02 = 42.010600 [^ 1</span><span class="c0">&nbsp;</span><span class="c10">(protein N-terminal acetylation)</span></p><p class="c4 c26"><span class="c0 c24"># variable_mod_03 = 79.966330 STY 3 </span><span class="c10">(for phosphorylation, commented out with &lsquo;#&rsquo; because its not needed)</span></p><p class="c4 c26"><span class="c0 c24"># variable_mod_04 = -17.026500 nQnC 1</span><span class="c0">&nbsp;</span><span>(for pyro-Glu or loss of ammonia at peptide N-terminal, commented out because its not needed)</span></p><p class="c4 c26"><span class="c0 c24"># variable_mod_05 = -18.010600 nE 1</span><span class="c0">&nbsp;</span><span class="c10">(for glutamic acid at the N-terminal, &nbsp;commented out because its not needed)</span></p><p class="c4 c26"><span class="c0 c24">variable_mod_06 = 229.162930 n^ 1 </span><span class="c10">(TMT on peptide N-terminus)</span></p><p class="c4 c26"><span class="c0 c24">variable_mod_07 = 229.162930 S 1 </span><span class="c10">(TMT on serine))</span></p><p class="c2"><span class="c21 c0"></span></p><p class="c4"><span class="c0 c24">Fixed modifications</span><span class="c10">&nbsp;set static numbers for mass for various peptides</span></p><p class="c2"><span class="c10"></span></p><p class="c4 c26"><span class="c21 c0">add_Q_glutamine = 0.000000</span></p><p class="c4 c26"><span class="c21 c0">add_K_lysine = 229.162930</span></p><p class="c4 c26"><span class="c21 c0">add_E_glutamic_acid = 0.000000</span></p><p class="c2 c26"><span class="c0 c16"></span></p><p class="c4"><span class="c10">For TMTpro, fixed modifications all become 304.207146</span></p><p class="c4"><span class="c10">For label-free, comment out variable mods and put fixed lysine mass to 0.0000</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">Another critical parameter is instrument resolution. Assuming a recent Orbitrap / Q-Exactive, we run in high-high resolution mode (precursors and fragments), which looks like this:</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c14 c0">precursor_mass_lower = -10 &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Lower bound of the precursor mass window.</span></p><p class="c4"><span class="c14 c0">precursor_mass_upper = 10 &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Upper bound of the precursor mass window.</span></p><p class="c4"><span class="c14 c0">precursor_mass_units = 1 &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Precursor mass tolerance units (0 for Da, 1 for ppm).</span></p><p class="c4"><span class="c14 c0">data_type = 0 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Data type (0 for DDA, 1 for DIA, 2 for gas-phase fractionation DIA).</span></p><p class="c4"><span class="c14 c0">precursor_true_tolerance = 10 &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# True precursor mass tolerance (window is +/- this value).</span></p><p class="c4"><span class="c14 c0">precursor_true_units = 1 &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# True precursor mass tolerance units (0 for Da, 1 for ppm).</span></p><p class="c4"><span class="c14 c0">fragment_mass_tolerance = 10 &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Fragment mass tolerance (window is +/- this value).</span></p><p class="c4"><span class="c14 c0">fragment_mass_units = 1 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Fragment mass tolerance units (0 for Da, 1 for ppm).</span></p><p class="c4"><span class="c0 c14">calibrate_mass = 2 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Perform mass calibration (0 for OFF, 1 for ON, 2 for ON and find optimal parameters).</span></p><p class="c4"><span class="c14 c0">use_all_mods_in_first_search = 0&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Use all variable modifications in the first search (0 for No, 1 for Yes).</span></p><p class="c4"><span class="c14 c0">decoy_prefix = rev_ &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Prefix of the decoy protein entries. Used for parameter optimization only.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">If we run any LTQ-Orbitrap (these are high-low), so we change fragment mass tolerance to 0.5Da. The rest of the parameters remain the same.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c14 c0">fragment_mass_tolerance = 0.5 &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Fragment mass tolerance (window is +/- this value).</span></p><p class="c4"><span class="c0 c31">fragment_mass_units = 0 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# Fragment mass tolerance units (0 for Da, 1 for ppm).</span></p><p class="c2"><span class="c10"></span></p><h2 class="c44" id="h.1gl7rvyauw8k"><span class="c35">3c. QC of data</span></h2><p class="c4"><span>Andy has made some</span><span>&nbsp;</span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://github.com/VibrantStarling/make-protein-atlas/tree/main/python-qc-scripts&amp;sa=D&amp;source=editors&amp;ust=1751561388009967&amp;usg=AOvVaw3cvmx_cmCYE-lHXTPEh3zW">python QC scripts</a></span><span>&nbsp;</span><span class="c10">that filter data and convert them into tsv format. In order to run these, you need to logged into the login node have run:</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c0 c24">module load python</span><span class="c21 c43">&nbsp;</span></p><p class="c4"><span class="c21 c0">pip3 install pandas matplotlib numpy</span></p><p class="c2"><span class="c16 c0"></span></p><p class="c4"><span class="c10">You should only have to do this once, but sometimes updates to the login node&rsquo;s Python might reset this. Consider setting up a conda environment.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">After running the python scripts, this extracts the peptide-spectrum match (PSM) data from the final pepXML file in the pipeline, into various CSV files, and applying FDR thresholding - the final file is: PXD0XXXXX_interact-ipro.pep_thresholded.tsv. A stats file is also produced PXD0XXXXX_psm_stats.tsv, which produces data like this:</span></p><p class="c2"><span class="c10"></span></p><table class="c42"><tr class="c15"><td class="c40" colspan="1" rowspan="1"><p class="c4"><span class="c25 c23">raw_file</span></p></td><td class="c1" colspan="1" rowspan="1"><p class="c4"><span class="c23 c25">spectra_count</span></p></td><td class="c8" colspan="1" rowspan="1"><p class="c4"><span class="c25 c23">psm_count</span></p></td><td class="c27" colspan="1" rowspan="1"><p class="c4"><span class="c25 c23">recovery</span></p></td></tr><tr class="c15"><td class="c40" colspan="1" rowspan="1"><p class="c4"><span class="c9">20201215_CDPK1TimeCourse_TMTPro1_WP_DMSO_fxn1.raw</span></p></td><td class="c1" colspan="1" rowspan="1"><p class="c4"><span class="c9">39076</span></p></td><td class="c8" colspan="1" rowspan="1"><p class="c4"><span class="c9">13058</span></p></td><td class="c27" colspan="1" rowspan="1"><p class="c4"><span class="c9">0.334169311</span></p></td></tr><tr class="c15"><td class="c40" colspan="1" rowspan="1"><p class="c4"><span class="c9">20201215_CDPK1TimeCourse_TMTPro1_WP_DMSO_fxn3.raw</span></p></td><td class="c1" colspan="1" rowspan="1"><p class="c4"><span class="c9">42884</span></p></td><td class="c8" colspan="1" rowspan="1"><p class="c4"><span class="c9">13547</span></p></td><td class="c27" colspan="1" rowspan="1"><p class="c4"><span class="c9">0.315898703</span></p></td></tr><tr class="c15"><td class="c40" colspan="1" rowspan="1"><p class="c4"><span class="c9">20201215_CDPK1TimeCourse_TMTPro1_WP_DMSO_fxn4.raw</span></p></td><td class="c1" colspan="1" rowspan="1"><p class="c4"><span class="c9">45142</span></p></td><td class="c8" colspan="1" rowspan="1"><p class="c4"><span class="c9">13986</span></p></td><td class="c27" colspan="1" rowspan="1"><p class="c4"><span class="c9">0.309822338</span></p></td></tr><tr class="c15"><td class="c40" colspan="1" rowspan="1"><p class="c4"><span class="c9">20201211_CDPK1TimeCourse_TMTPro1_WP_ZAP_fxn1.raw</span></p></td><td class="c1" colspan="1" rowspan="1"><p class="c4"><span class="c9">30469</span></p></td><td class="c8" colspan="1" rowspan="1"><p class="c4"><span class="c9">9240</span></p></td><td class="c27" colspan="1" rowspan="1"><p class="c4"><span class="c9">0.30325905</span></p></td></tr><tr class="c15"><td class="c40" colspan="1" rowspan="1"><p class="c4"><span class="c9">20201211_CDPK1TimeCourse_TMTPro1_WP_ZAP_fxn4.raw</span></p></td><td class="c1" colspan="1" rowspan="1"><p class="c4"><span class="c9">42892</span></p></td><td class="c8" colspan="1" rowspan="1"><p class="c4"><span class="c9">12905</span></p></td><td class="c27" colspan="1" rowspan="1"><p class="c4"><span class="c9">0.300871957</span></p></td></tr></table><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c0 c24">psm_count </span><span class="c10">is count of PSMs at 1% FDR, </span></p><p class="c4"><span class="c0 c24">spectra_count </span><span class="c10">is count of total spectra, </span></p><p class="c4"><span class="c0 c24">recovery</span><span class="c10">&nbsp;is the 1% FDR count / total count</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span>If the analysis has gone well, we are looking at recovery of say 25-60% PSMs per raw file (and ideally a large volume of spectra searched). Sometimes the proportion drops really low (say &lt;10%) - this is usually because we have put the search settings wrong, or perhaps it is low quality data. </span></p><hr style="page-break-before:always;display:none;"><h1 class="c22" id="h.q99yog46sgqg"><span class="c18"></span></h1><h1 class="c30" id="h.imhqa9nobeaw"><span class="c18">4. Combining and collapsing data</span></h1><p class="c4"><span>Run </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://github.com/VibrantStarling/make-protein-atlas/blob/main/slurm-and-shell-scripts/run-combine-and-threshold-BATCH.sh&amp;sa=D&amp;source=editors&amp;ust=1751561388017163&amp;usg=AOvVaw1VqE365u84icu60KFqxcKM">run-combine-and-threshold-BATCH.sh</a></span><span>&nbsp;t</span><span class="c10">o combine, collapse, and threshold all the individual PXD files by peptide.</span></p><p class="c4"><span><br></span><span class="c0 c24">sbatch run-combine-and-threshold-BATCH.sh pxdlist.txt outsuffix</span></p><h1 class="c30" id="h.wgxmc6v1wcpe"><span class="c18">5. Visualising peptides on a genome</span></h1><p class="c4"><span class="c17">ANDY: Ideally, I would like to convert peptide-level data to BED or BAM format for visualising directly in genome browsers, like JBrowse. This is technically fairly straightforward, since we could get the GFF of gene models from which proteins are derived. From proteomics results, we get the position of the peptide within a protein, and then need an extra step to figure out the corresponding genomic position - including mapping some peptides across intron junctions. The algorithm is a little fiddly but not really that complex, but I don&rsquo;t currently have a code base I am happy with. This is a task for someone to take on&hellip;</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">This has yet to be fully hashed out. In theory the steps might include:</span></p><ol class="c7 lst-kix_51wdrdk2fq9o-0 start" start="1"><li class="c3 li-bullet-0"><span class="c10">Find a way to translate pepXML to a gff.</span></li><li class="c3 li-bullet-0"><span class="c10">Map peptides to the genome and generate a SAM file</span></li><li class="c3 li-bullet-0"><span class="c10">Use samtools to convert the SAM file to BAM, then index the BAM file</span></li><li class="c3 li-bullet-0"><span>Use bedtools intersect to identify shared regions between the gff and bam file (</span><span class="c12"><a class="c19" href="https://www.google.com/url?q=http://quinlanlab.org/tutorials/bedtools.html%23bedtools-intersect&amp;sa=D&amp;source=editors&amp;ust=1751561388019636&amp;usg=AOvVaw2y0vIqRJycK_d3r0JQ2m4J">http://quinlanlab.org/tutorials/bedtools.html#bedtools-intersect</a></span><span class="c10">)</span></li><li class="c3 li-bullet-0"><span>Create a bigwig file and use that to visualise continuous protein data </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionV/lessons/10_data_visualization.html&amp;sa=D&amp;source=editors&amp;ust=1751561388020387&amp;usg=AOvVaw02GM87MA4gYkxwV84--HjW">https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/sessionV/lessons/10_data_visualization.html</a></span><span class="c10">&nbsp;</span></li></ol><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">You can load a BED or BAM file in to IGV to visualise the data alongside genome data</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span>proBAMsuite, a possibly useful tool? </span><span class="c12"><a class="c19" href="https://www.google.com/url?q=https://www.mcponline.org/article/S1535-9476(20)33669-0/pdf&amp;sa=D&amp;source=editors&amp;ust=1751561388021285&amp;usg=AOvVaw1J4jhVsjACTleg5zs79uTc">https://www.mcponline.org/article/S1535-9476(20)33669-0/pdf</a></span><span class="c10">&nbsp;</span></p><p class="c2"><span class="c10"></span></p><hr style="page-break-before:always;display:none;"><h1 class="c22" id="h.hurhofdh7evo"><span class="c18"></span></h1><h1 class="c30" id="h.oxvtznduwpe0"><span class="c18">Running the pipeline</span></h1><p class="c4"><span class="c28 c47">process-pxd.sh</span><span class="c16 c23">&nbsp;combines all the necessary steps to download, process, and qc a set of PXD files. You still have to assemble the PXD files and fasta database yourself, but the rest should be automated.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c10">It allows you to run each stage of the pipeline separately, so if you need to add a file, or just rerun the postprocessing step, you can do it without reprocessing every single file.<br><br>You can download a repository with all the necessary directories and directory structures from </span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c21 c0">git clone https://github.com/VibrantStarling/make-protein-atlas.git</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c16 c23">General usage:</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 105.33px;"><img alt="" src="images/image4.png" style="width: 624.00px; height: 105.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4"><span class="c10">If you do not specify any of the flags (-D, -M, or -F), then it will run the whole pipeline by default.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c36 c23 c29">This script is functional, but not optimised, so you may need a specific file layout for it. However, in theory, you could have one folder for all the scripts and just supply that one file to each of the directory options (-q, -s, -p)</span></p><p class="c2"><span class="c36 c23 c29"></span></p><p class="c4"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 429.00px; height: 86.00px;"><img alt="" src="images/image8.png" style="width: 429.00px; height: 86.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c16 c23">You will need:</span></p><ol class="c7 lst-kix_xf4ne1qm8zsq-0 start" start="1"><li class="c3 li-bullet-0"><span class="c28">A PXD list. </span><span>A</span><span>&nbsp;list of PXD accessions and the type of data in them. Accession and type are delimited by a tab, and each row has a separate entry.</span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 227.00px; height: 145.00px;"><img alt="" src="images/image7.png" style="width: 227.00px; height: 145.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></li></ol><p class="c4 c26"><span>Type defines which MS-fragger params file will be used. It will always look for the pattern of </span><span class="c0 c24">fragger_${type}.params</span><span>, where ${type} could be filled in with </span><span class="c0 c24">LM </span><span>to make </span><span class="c0 c24">fragger_LM.params</span><span class="c10">. If you make a new .params file, make sure it follows that naming convention.</span></p><ol class="c7 lst-kix_xf4ne1qm8zsq-0" start="2"><li class="c3 li-bullet-0"><span class="c28">A database fasta file.</span><span class="c10">&nbsp;A fasta file of protein information to search against. This is a collection of your organism of interest, reverse reads, and contaminants.</span></li></ol><p class="c4 c26"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 133.00px;"><img alt="" src="images/image2.png" style="width: 624.00px; height: 353.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><ol class="c7 lst-kix_xf4ne1qm8zsq-0" start="3"><li class="c3 li-bullet-0"><span>A directory that contains all the </span><span class="c16 c23">python QC scripts</span></li></ol><p class="c4 c26"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 86.67px;"><img alt="" src="images/image6.png" style="width: 624.00px; height: 86.67px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><ol class="c7 lst-kix_xf4ne1qm8zsq-0" start="4"><li class="c3 li-bullet-0"><span>A directory containing </span><span class="c28">additional slurm and bash scripts. </span><span class="c36 c29 c43">(Yours may look cleaner than this and won&rsquo;t have four different TPP shell scripts)</span></li></ol><p class="c4 c26"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 109.33px;"><img alt="" src="images/image1.png" style="width: 624.00px; height: 109.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><ol class="c7 lst-kix_xf4ne1qm8zsq-0" start="5"><li class="c3 li-bullet-0"><span class="c28">A directory containing MS fragger parameter files,</span><span class="c10">&nbsp;with names that correspond to the type specified in the PXD-list</span></li></ol><p class="c4 c26"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 94.67px;"><img alt="" src="images/image5.png" style="width: 624.00px; height: 94.67px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c2"><span class="c10"></span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span>Feed the two files and three directories to the</span><span class="c0 c24">&nbsp;process-pxd.sh </span><span class="c10">script and it will run the whole pipeline by default. It will download and produce directories in your current directory.</span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c0 c21">bash process-pxd.sh -l pxd_list.txt -b path/to/database.fasta -q path/to/python-qc-directory -s path/to/slurm-and-bash-script-directory -p path/to/MS-fragger-params-directory</span></p><p class="c2"><span class="c21 c0"></span></p><p class="c4"><span class="c10">If you need to run the download (-D), mzml conversion (-M), or fragger search and postprocessing steps (-F) separately, add the relevant flag. </span></p><p class="c2"><span class="c10"></span></p><p class="c4"><span class="c16 c23">After you have run the pipeline go to step 4, combining and collapsing data</span></p></body></html>